services/api/service.go

```diff
@@ -1335,9 +1333,47 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		}
 	}()
 
+	// Get the current (in-memory) top bid value
+	api.builderTopBidValueLock.RLock()
+	localTopBidValue := api.builderTopBidValue[payload.ParentHash()]
+	api.builderTopBidValueLock.RUnlock()
+
+	// Without cancellations, discard lower or similar value submissions to previous top bid
+	if !isCancellationEnabled {
+		// Check the local top bid value first
+		if localTopBidValue != nil && payload.Value().Cmp(localTopBidValue) < 1 {
+			log.Info("rejecting submission because it is lower or equal to the top bid (local)")
+			w.WriteHeader(http.StatusOK)
+			return
+		}
+
+		// Then check the top bid value in redis
+		topBidValue, err := api.redis.GetTopBidValue(payload.Slot(), payload.ParentHash(), payload.ProposerPubkey())
+		if err != nil {
+			log.WithError(err).Error("failed to get top bid value from redis")
+		} else if topBidValue != nil {
+			log = log.WithField("topBidValue", topBidValue.String())
+
+			// Save locally for later in-memory lookup
+			if localTopBidValue == nil || localTopBidValue.Cmp(topBidValue) < 1 {
```

Comment:
Do we need to update the builderTopBidValue if it's the same value as `localTopBidValue` ? 

I'm wondering if we could do `localTopBidValue.Cmp(topBidValue) < 0` instead

Reply:
removed the comparison, it should always overwrite, even if lower